test/example.js

```diff
@@ -1,20 +1,28 @@
 'use strict';
 
-const methodName = (
-  // Method caption
-  // First line description
-  // second line of multiline method description
-  a1, // number, comment
-  // multiline comment
-  // for first argument
-  a2, // array of string
-  a3 // comment
-  // Returns: string, comment
-  // Hint: message
-  // Example: methodName(5, [2, 3], 'f')
-  // Result: '52,3f'
-) => {
+// Method caption
+// First line description
+// second line of multiline method description
+//   a1 - <number>, comment
+//     multiline comment, something
+//     for the first argument
+//   a2 - <string[]>, comments
+//   a3 - comment
+//   fn - <function>, fn(a, b, c), description
+//     a - <object>, also description
+//     b - <boolean> | <number>
+//     c - <array>, comment
+// Returns: <string>, comment
+// Hint: message
```

Comment:
Why do we need `hint`?
I mean, why don't we just put anything that might go in `hint` directly to the description up top?

Reply:
I don't know either though I suspect there can be something not related to usage like information about how internals work. If someone comes up with a better explanation, I'd be glad to hear it. As for now I think we should leave it here.